packages/compass-e2e-tests/index.ts

```diff
@@ -271,15 +279,20 @@ async function main() {
     return index >= minGroupIndex && index <= maxGroupIndex;
   });
 
-  console.info('Test files:', rawTests);
+  console.info('Test files:', tests);
 
   // The only test file that's interested in the first-run experience (at the
   // time of writing) is time-to-first-query.ts and that happens to be
   // alphabetically right at the end. Which is fine, but the first test to run
   // will also get the slow first run experience for no good reason unless it is
   // the time-to-first-query.ts test.
   // So yeah.. this is a bit of a micro optimisation.
-  const tests = [FIRST_TEST, ...rawTests.filter((t) => t !== FIRST_TEST)];
+  for (let i = 0; i < tests.length; ++i) {
+    if (tests[i] === FIRST_TEST) {
+      [tests[i], tests[0]] = [tests[0], tests[i]];
+      break;
+    }
+  }
```

Comment:
Why is this changed?

Reply:
Because I added the test filtering feature, and this line previously was always including the FIRST_TEST.

Reply:
Ah, gotcha, was kinda hard to understand from the code. Maybe better to be a bit more explicit about it?

```suggestion
  const shouldRunAllTests = e2eTestFilter === '*';
  const tests = shouldRunAllTests ? <logic to move FIRST_TEST to the front> : <just exactly what glob returned>
```

Reply:
Sure, will update

Reply:
Actually, looking at this again, I think it's more like we want to move time-to-first-query.test.ts to the front only if it is one of the tests we will run (it may still be included if a filter is used).

So technically, we need to check if it's in the full array, and that point we are already iterating the array, so we probably don't need to disambiguate the two cases here.

Reply:
Sure, if the intent is to be able to pick up more than one file, this makes sense. Can we rewrite this using a `sort` then instead of in place swapping so that the code is readable and the intent is clear?